### PR TITLE
Basket working with router now

### PR DESCRIPTION
**Pull request by Maximca2**

### CODE REVIEW

**Comment by Maximca2** [comment]
This is task 4

Move the shopping cart to a separate page

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import React from 'react';
 import { Container } from 'react-bootstrap';
 
-import MainPage from './pages/MainPage';
+import MainRoute from './pages/MainRoutePage/MainRoute';
```
Comment (Nerevand) [inline src/App.js]: in this branch u also have  MainPage folder. Remove it if u don't use

Comment (Nerevand) [reply src/App.js]: inside `pages ` folder u have a real page, but not a wrapper with routes and header. U can move this logic in `App.js` file

File: src/pages/Basket/Basket.jsx [hunk 2]
```diff
@@ -1,41 +1,32 @@
...
-                const {name,img,price} = it.product;
-                return (
+      <div className={style.box}>
+        <div className={style.basket}>
+          <div className={style.basket__list}>
+            {!curProducts.length
+              ? "Корзина пуста"
+              : curProducts.map((it, i) => (
                   <div key={i} className={style.basket__card}>
                     <div className="">
```
Comment (Nerevand) [inline src/pages/Basket/Basket.jsx]: empty `className`. Remove it or add real class

File: src/pages/Basket/Basket.jsx [hunk 3]
```diff
@@ -1,41 +1,32 @@
...
-                      <img src={img} alt="curimg" />
+                      <img src={it.product.img} alt="curimg" />
                     </div>
-                    {name}
-                    <div className="">{price}</div>
+                    {it.product.name}
+                    <div className="">{it.product.price}</div>
```
Comment (Nerevand) [inline src/pages/Basket/Basket.jsx]: empty `className`. Remove it or add real class

File: src/pages/Basket/Basket.jsx [hunk 4]
```diff
@@ -1,41 +1,32 @@
...
-            ? "Корзина пуста"
-            : curProducts.map((it, i) => {
-                const {name,img,price} = it.product;
-                return (
+      <div className={style.box}>
+        <div className={style.basket}>
+          <div className={style.basket__list}>
+            {!curProducts.length
+              ? "Корзина пуста"
+              : curProducts.map((it, i) => (
```
Comment (Nerevand) [inline src/pages/Basket/Basket.jsx]: use `destructuring`

File: src/pages/Basket/Basket.module.scss [hunk 5]
```diff
@@ -41,7 +41,7 @@
 
   padding: 0 20px;
   pointer-events: auto;
-//   position: absolute;
+  //   position: absolute;
```
Comment (Nerevand) [inline src/pages/Basket/Basket.module.scss]: remove commented lines for that file

File: src/pages/MainPage/MainPage.jsx [hunk 6]
```diff
@@ -1,31 +1,23 @@
...
-
 import style from "./MainPage.module.scss";
 
 const MainPage = () => {
   const dispatch = useDispatch();
   const products = useSelector((state) => state.items.it);
-  const [show, setShow] = useState(true);
   const curProducts = useSelector((state) => state.items.basket);
-
+  const ADD_TO_BASKET ='ADD_TO_BASKET'
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx]: `ADD_TO_BASKET`  move it to actions file

File: src/pages/MainPage/MainPage.jsx [hunk 7]
```diff
@@ -66,20 +57,18 @@ const MainPage = () => {
             <nav>
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx:59]: in this file u have empty classNames. FIx it please

File: src/pages/MainRoutePage/MainRoute.jsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import { NavLink, Route, Routes } from "react-router-dom";
+
+import MainPage from "../MainPage/MainPage";
+import Basket from "../Basket/Basket";
+import NotFound from "../Not found/Not found";
+
+import style from "../MainRoutePage/Main.route.module.scss";
+
+const MainRoute = () => {
+    const ROOT_TO_MAINPAGE = '/mainpage';
```
Comment (Nerevand) [inline src/pages/MainRoutePage/MainRoute.jsx]: `ROOT_TO_MAINPAGE `and `ROOT_TO_BASKET` u can create it outside a component

File: src/pages/Not found/Not found.jsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+const pageNotFound =
+  "https://internetdevels.com/sites/default/files/public/blog_preview/404_page_cover.jpg";
+
+const NotFound = () => {
+  return (
+    <div>
+      <div className="">
```
Comment (Nerevand) [inline src/pages/Not found/Not found.jsx]: fix empty className``

File: src/App.js [hunk 10]
```diff
@@ -1,16 +1,53 @@
 import React from 'react';
 import { Container } from 'react-bootstrap';
+import { NavLink, Route, Routes } from "react-router-dom";
 
-import MainPage from './pages/MainPage';
+import MainPage from "./pages/MainPage/MainPage";
+import Basket from "./pages/Basket";
+import NotFound from "./pages/Not found/Not found";
+import { ROOT_TO_MAINPAGE,ROOT_TO_BASKET } from './Routes/routes';
```
Comment (Nerevand) [inline src/App.js]: `Routes` folder it is not a `react component`. Naming should be in lower case

File: src/pages/Basket/Basket.jsx [hunk 11]
```diff
@@ -1,51 +1,43 @@
...
+          <div className={style.basket__list}>
+            {!curProducts.length
+              ? "Корзина пуста"
+              : curProducts.map(({ product }, i) => {
+                  const { name, price, img } = product;
+                  return (
+                    <div key={i} className={style.basket__card}>
                       <img src={img} alt="curimg" />
+                      {name}
+                      <div className="">{price}</div>
```
Comment (Nerevand) [inline src/pages/Basket/Basket.jsx]: empty `className`

File: src/pages/MainPage/MainPage.jsx [hunk 12]
```diff
@@ -1,31 +1,24 @@
-import React, { useState } from "react";
+import React from "react";
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx:1]: still exist here empty `classNames`

**Review by Nerevand — APPROVED** [review]

**Pull request merged**